src/palette.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+)
+
+type Palette map[string]string
+
+const (
+	paletteKeyError          = "palette: requested color %s does not exist in palette of colors %s"
+	paletteRecursiveKeyError = "palette: resolution of color %s returned palette reference %s; recursive references are not supported"
+)
+
+var (
+	paletteColorPrefixes = [...]string{"palette:", "p:"}
```

Comment:
I would remove this and only use `p:`

Reply:
Removed "palette:" from valid references.